### PR TITLE
Implement csi sanity check

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
Kubernetes provides a few test suites you can run against your CSI driver to make sure it properly implements the CSI spec. One of those is the [csi sanity](https://github.com/kubernetes-csi/csi-test/tree/master/pkg/sanity) test. It is essentially a supplement unit test that tests that we appropriately respond to various inputs from the csi containers like the csi-provisioner. It is one step in ensuring we support the minimum needed to be a proper csi driver.

* Added make target so we can run this in a CI lane calling the target
* Implemented missing features that were required to pass the sanity test.
* Added a lot of input validation for the various CSI request objects so that would pass the sanity test.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: sanity/sanity_test.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+	if k.vmiMap[vmKey] == nil {
+		return fmt.Errorf("VM %s/%s not found", namespace, vmName)
+	}
+	delete(k.hotpluggedMap, hotPlugRequest.Name)
+	return nil
+}
+
+func (m *fakeKubeVirtClient) List() ([]byte, error) {
+	ds := make([]device, 0)
+	for _, value := range m.hotpluggedMap {
```
Comment (brybacki) [inline sanity/sanity_test.go:205]: kubevirt as `m`?

Comment (awels) [reply sanity/sanity_test.go:205]: I had it as a different struct, good catch should be k like the rest.

**Comment by brybacki** [comment]
/test pull-csi-driver-e2e-k8s-1.23
          s: "Pod in phase Pending, expected Running",

**Comment by brybacki** [comment]
it is great to have the k8s csi driver sanity tests running with standard unit test suite! 

as far as I understand this code, it looks good

**Comment by awels** [comment]
The sanity lane is passing, should be good to go.

**Comment by brybacki** [comment]
/test pull-csi-driver-sanity-test

```
        s: "Connection timed out",
```

**Comment by brybacki** [comment]
@davidvossel wdyt?

**Review by davidvossel — APPROVED** [review]
/approve

nice! thanks

**Pull request merged**